lib/facility.rb

```diff
@@ -1,14 +1,37 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
   def initialize(info)
     @name = info[:name]
     @address = info[:address]
     @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
   def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
```

Comment:
register_vehicle method is the main part of Vehicle registration portion.  Employs several helper methods from Vehicle class